code/messaging/src/main/java/com/adobe/marketing/mobile/messaging/ContentCardMapper.kt

```diff
@@ -0,0 +1,60 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.messaging
+
+import com.adobe.marketing.mobile.util.StringUtils
+
+/**
+ * Class to store a mapping between valid [ContentCardSchemaData] and unique proposition id's.
+ */
+class ContentCardMapper private constructor() {
+    private val contentCardSchemaDataMap: MutableMap<String, ContentCardSchemaData> = HashMap()
+
+    companion object {
+        @JvmStatic
+        val instance: ContentCardMapper by lazy { ContentCardMapper() }
+    }
+
+    /**
+     * Stores the [ContentCardSchemaData] for the given proposition id.
+     *
+     * @param contentCardSchemaData the [ContentCardSchemaData] to store
+     */
+    fun storeContentCardSchemaData(contentCardSchemaData: ContentCardSchemaData) {
+        if (contentCardSchemaData.parent.propositionReference == null) {
+            return
+        }
+        contentCardSchemaDataMap[contentCardSchemaData.parent.proposition.uniqueId] = contentCardSchemaData
+    }
+
+    /**
+     * Removes the [ContentCardSchemaData] for the given proposition id.
+     *
+     * @param propositionId the proposition id to remove the [ContentCardSchemaData] for
+     */
+    fun removeContentCardSchemaDataFromMap(propositionId: String) {
+        contentCardSchemaDataMap.remove(propositionId)
+    }
+
+    /**
+     * Returns the [ContentCardSchemaData] for the given proposition id.
+     *
+     * @param propositionId the proposition id to retrieve the [ContentCardSchemaData] for
+     * @return the [ContentCardSchemaData] for the given proposition id, or null if not found
+     */
+    fun getContentCardSchemaDataForPropositionId(propositionId: String): ContentCardSchemaData? {
```

Comment:
Was chatting with Spoorthi and it seems like we only need this utility method to be public. Can we rename this to
`fun getContentCardSchemaData(propositionId: String)` and make only this public and switch rest of them to internal.